ckks/bootstrapping/bootstrap.go

```diff
@@ -0,0 +1,122 @@
+package bootstrapping
+
+import (
+	"github.com/ldsec/lattigo/v2/ckks"
+	"github.com/ldsec/lattigo/v2/ring"
+	"math"
+)
+
+// Bootstrapp re-encrypt a ciphertext at lvl Q0 to a ciphertext at MaxLevel-k where k is the depth of the bootstrapping circuit.
+// If the input ciphertext level is zero, the input scale must be an exact power of two smaller or equal to round(Q0/2^{10}).
+// If the input ciphertext is at level one or more, the input scale does not need to be an exact power of two as one level
+// can be used to do a scale matching.
+func (btp *Bootstrapper) Bootstrapp(ctIn *ckks.Ciphertext) (ctOut *ckks.Ciphertext) {
```

Comment:
Since the function nicely short now, I would put everything in bootstrapper.go.

Reply:
agree